stan/math/prim/prob/von_mises_cdf.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef STAN_MATH_PRIM_PROB_VON_MISES_CDF_HPP
+#define STAN_MATH_PRIM_PROB_VON_MISES_CDF_HPP
+
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/prob/normal_cdf.hpp>
+#include <stan/math/prim/fun/modified_bessel_first_kind.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+namespace internal {
+
+template <typename T_x, typename T_k, typename T_p>
+return_type_t<T_x, T_k, T_p> von_mises_cdf_series(const T_x& x, const T_k& k,
+                                                  const T_p& p) {
+  using stan::math::cos;
+  using stan::math::sin;
+
+  double pi = stan::math::pi();
+  auto s = sin(x);
+  auto c = cos(x);
+  auto sn = sin(p * x);
+  auto cn = cos(p * x);
+
+  return_type_t<T_x, T_k, T_p> R = 0;
+  return_type_t<T_x, T_k, T_p> V = 0;
+
+  int n;
+  for (n = p - 1; n > 0; n--) {
+    auto sn_tmp = sn * c - cn * s;
+    cn = cn * c + sn * s;
+    sn = sn_tmp;
+    R = 1 / (2.0 * n / k + R);
+    V = R * (sn / n + V);
+  }
```

Comment:
Can you do `for (int n = p - 1; n > 0; n--) {`?

Also optional, it's usually easier to read code that's done as a forward iterator instead of reverse. idk if that's possible here but would be nice if so

Reply:
Do ```--n``` too. I think we favor that style?

Reply:
it's nanoseconds faster but I'm indifferent either way

Reply:
I feel the same.